File: code-bdd/RxObserverExample.swift
import Foundation
import RxSwift

class RxObserverExample {

    let section: BehaviorSubject&lt;Section> =
            BehaviorSubject&lt;Section>(value: Section("++", [Paragraph](), [Section]()))

    private (set) var details: String = "H:++, P:++, S:++"  // private setter, public getter

    init() {
        section.subscribe(onNext: { it in                   // Rx subscription
            print("onNext() section heading = \(it.heading)")
            self.details =
            "H:\(it.heading ?? "--"), " +          // conditional assignment ??
            "P:\(it.paragraphs.count), " +
            "S:\(it.sections?.count ?? -1)"    // conditional assignment ??
        })
    }

    var description: String {  // getter
        get {
            return "Details for section :: " + details
        }
    }
}

File: code-bdd/RxObserverExample.kt
package org.mobilerosetta.mobilerosetta

import io.reactivex.subjects.BehaviorSubject

class RxObserverExample {

    val initialValue = Section("++", listOf(), listOf())
    val section = BehaviorSubject.create&lt;Section>()

    var details: String = "H:++, P:++, S:++"
        private set
    
    init {
        section.onNext(initialValue)
        section.subscribe {
            println("onNext() section heading = ${it.heading}")
            details =
                    "H:${it.heading
                            ?: "--"}, " +                // conditional assignment, elvis symbol
                    "P:${it.paragraphs.count()}, " +
                    "S:${it.sections?.count()
                            ?: -1}"      // optional method call, optional assignment, elvis symbol
        }
    }

    val description: String
        get() = "Details for section :: " + details  // getter shorthand
}
